packages/astro/src/content/utils.ts

```diff
@@ -146,33 +178,45 @@ export function getContentEntryConfigByExtMap(settings: Pick<AstroSettings, 'con
 	return map;
 }
 
-export class NoCollectionError extends Error {}
-
-export function getEntryInfo(
-	params: Pick<ContentPaths, 'contentDir'> & {
-		entry: string | URL;
-		allowFilesOutsideCollection?: true;
+export function getEntryCollectionName({
```

Comment:
Refactored wacky `NoCollectionError` and function overloading to a separate query for collection names. This hopefully cleans up some readability problems!